src/updater/updater.cpp

```diff
@@ -149,446 +95,568 @@ void CheckWeightsConsistency(
-    util::Log() << "Reading " << graph_header.number_of_edges << " edges from the edge based graph";
-
-    auto segment_speed_lookup = csv::readSegmentValues(config.segment_speed_lookup_paths);
-    auto turn_penalty_lookup = csv::readTurnValues(config.turn_penalty_lookup_paths);
-
-    std::vector<extractor::QueryNode> internal_to_external_node_map;
-    extractor::SegmentDataContainer segment_data;
-
-    const auto maybe_load_internal_to_external_node_map = [&] {
-        if (!update_edge_weights)
-            return;
-
-        storage::io::FileReader nodes_file(config.node_based_graph_path,
-                                           storage::io::FileReader::HasNoFingerprint);
-
-        nodes_file.DeserializeVector(internal_to_external_node_map);
-    };
-
-    const auto maybe_load_geometries = [&] {
-        if (!update_edge_weights)
-            return;
-
-        extractor::io::read(config.geometry_path, segment_data);
-    };
-
-    // Folds all our actions into independently concurrently executing lambdas
-    tbb::parallel_invoke(maybe_load_internal_to_external_node_map, maybe_load_geometries);
+        BOOST_ASSERT(segment_data_backup);
 
-    if (update_edge_weights)
-    {
-        // Here, we have to update the compressed geometry weights
-        // First, we need the external-to-internal node lookup table
-
-        // Now, we iterate over all the segments stored in the StaticRTree, updating
-        // the packed geometry weights in the `.geometries` file (note: we do not
-        // update the RTree itself, we just use the leaf nodes to iterate over all segments)
-        using LeafNode = util::StaticRTree<extractor::EdgeBasedNode>::LeafNode;
-
-        using boost::interprocess::mapped_region;
-
-        auto region = mmap_file(config.rtree_leaf_path.c_str(), boost::interprocess::read_only);
-        region.advise(mapped_region::advice_willneed);
-        BOOST_ASSERT(is_aligned<LeafNode>(region.get_address()));
-
-        const auto bytes = region.get_size();
-        const auto first = static_cast<const LeafNode *>(region.get_address());
-        const auto last = first + (bytes / sizeof(LeafNode));
-
-        // vector to count used speeds for logging
-        // size offset by one since index 0 is used for speeds not from external file
-        using counters_type = std::vector<std::size_t>;
-        std::size_t num_counters = config.segment_speed_lookup_paths.size() + 1;
-        tbb::enumerable_thread_specific<counters_type> segment_speeds_counters(
-            counters_type(num_counters, 0));
-        const constexpr auto LUA_SOURCE = 0;
-
-        tbb::parallel_for_each(first, last, [&](const LeafNode &current_node) {
-            auto &counters = segment_speeds_counters.local();
-            for (size_t i = 0; i < current_node.object_count; i++)
+        for (DirectionalGeometryID geometry_id = 0;
+             geometry_id < segment_data.GetNumberOfGeometries();
+             geometry_id++)
```

Comment:
`for (auto it = 0; it < fn(); ++it)`

computes `fn()` on every loop, why not

`for (auto it = 0, last = fn(); it < last; ++it)`

or

`irange(it, fn());`